### DISCUSSION TITLE
Is it possible to resize the toc in my documents?

### DISCUSSION THREAD

**Discussion by dorsalmo** [dotnet/docfx#7575] [Q&A]
I use "-t statictoc", which normally is great, but some of my toc elements are too long, and they don't wrap.

UPDATE:  This problem doesn't happen when I add "--serve" to my docfx command and view the document on http://localhost:8080/.  Too bad that's not an option.  I need anyone to be able to view them locally on their own PC.

![Capture](https://user-images.githubusercontent.com/6934037/130862828-5f5647ea-a19f-4dfc-882b-f2e9a2629919.JPG)

**Comment by kemeril** [comment]
Overwrite tha main.css file:
~\templates\staticdoc.override.partial\styles\main.css

Content:
@media screen and (min-width:1200px){
	.container {
		width:100%;
	}
	.sidefilter	 {
		width: 400px;
	}
	.sidetoc {
		width: 400px;
	}
	
	.article.grid-right {
		margin-left: 420px;
	}
}

**Reply by dorsalmo** [reply]
Thanks for responding.  There was no templates folder in my root docFx folder, so I created the one you indicated and added it to a new template section in docfx.json.  Unfortunately, it did not change the TOC size - even if I increase your 400s to 600s.  What could I be doing wrong?

**Comment by kemeril** [comment]
Sorry, you are right.
The default template is the "default" which requires to be hosted on a web server.
Therefor I use built in statictoc template. so no needed to be host just open ~\_site\index.htm file.

Each template may have different html layout so they need different css overriding.
The best you open the generated help in eg. Chrome browser and use it's developer tools to experiment with what is the best solution, which css overriding needed.
After that locate the main.css file under ~\site\ folder and define the css settings and test it by reloading the pages in the browser.
If you solution is passed as final step copy this css settings into the ~\templates\{template_name}.override.partial\styles\main.css file where the {template_name} shall be replaced by the used template name.
In my case staticdoc but in your case might be default.

Rebuild the help and test it again.

**Reply by dorsalmo** [reply]
Thanks again.  I'm having _some_ success.  I can alter the vertical size of the sidetoc but the horizontal size.  So, I thought I'd try specifying "overflow: scroll" to create a horizontal scroll bar, but it doesn't think it needs a scroll bar.  (Using "overflow: auto" confirms this - no scroll bar appears even though it is needed.)  I've read elsewhere that my size has to be fixed for this to work, but it is fixed.   I have the width set to a fixed amount.  I even set the height, though I don' really want to do so.  It didn't matter.  Are there children of sidetoc (or parents) that need to be altered for this to work?  I'm not getting very far with what seems like it should be easy.

**Reply by dorsalmo** [reply]
I ended up exporting the statictoc and default templates, determining which files were different, copying those to my template folder, and then making my change.  In addition, I needed to use both the default and my templates in my "docfx build -t" command because otherwise I kept getting errors about "no template processing document type" (see [this page](https://dotnet.github.io/docfx/tutorial/howto_create_custom_template.html)).  Maybe there was a better way to do it, but at least it's working now.
